code-challenges.test.js

```diff
@@ -23,10 +40,58 @@ const secretCodeWord3 = "Eccentric"
 
 // b) Create the function that makes the test pass.
 
+//Create a function called 'messageDecoder' that takes a string as a parameter
+//Use '.split('')' to turn string into an array with all letters at a different index
+//Loop through the array with '.map'
+//Use conditional statement and equality operator, if conditions are met convert "a" to 4, "e" to 3, "i" to 1, and "o" to 0, and all other letters left as is.
+// Use '.toLowerCase()' to account for capitilization,
+//Use '.join()' and '.toString' to return string
+
+const messageDecoder = (string) => {
+  array = string.split("").map((value) => {
+    if (value.toLowerCase() === "a") return "4";
+    else if (value.toLowerCase() === "e") return "3";
+    else if (value.toLowerCase() === "i") return "1";
+    else if (value.toLowerCase() === "o") return "0";
+    else return value;
+  });
+  return array.join("");
+};
+
```

Comment:
Nice job! As a challenge, how can you refactor this to be more efficient, not requiring a bunch of if / else if statements? Also, it's good practice to make sure you have a variable declaration. Last thing, I see that you wanted to store the logic inside `array` to return `array.join`, but you don't need to make `array` at all! You can just return the logic with `.join` attached to the end, just like how you chained `.split` and `.map`!